src/coreclr/jit/codegenxarch.cpp

```diff
@@ -9958,8 +9969,18 @@ unsigned CodeGen::genPopCalleeSavedRegistersFromMask(regMaskTP rsPopRegs)
 #ifdef TARGET_AMD64
     if ((rsPopRegs & RBM_R12) != 0)
     {
-        popCount++;
-        inst_RV(INS_pop, REG_R12, TYP_I_IMPL);
+#ifdef SWIFT_SUPPORT
+        static_assert_no_msg(RBM_SWIFT_ERROR == RBM_R12);
+        assert(compiler->lvaSwiftErrorArg == BAD_VAR_NUM);
+
+        // If this method returns an error argument in the Swift error register,
+        // we didn't push the register, and thus shouldn't pop it.
+        if (compiler->lvaSwiftErrorArg == BAD_VAR_NUM)
```

Comment:
Seems odd given the assert.

Reply:
Sorry I meant to remove that; doing so seems to fix the jitstressregs failure locally.

Reply:
Perhaps the caller shouldn't be passing it? Seems we could unify a bit of code by replacing some of the `RBM_INT_CALLEE_SAVED` occurrences in the backend with a function that takes the error return into account.

Reply:
That sounds better. Do you think it would make sense for this function to be a member of `RegSet` -- something like `RegSet::rsGetModifiedCalleeSavedRegsMask` -- since `RegSet` already has a `Compiler*` member variable to check for `lvaSwiftErrorArg`? Or would it be confusing if `RegSet::rsGetModifiedCalleeSavedRegsMask` sometimes doesn't include `REG_SWIFT_ERROR`, which is technically a callee-save register, even if it was modified?

Reply:
I think it would be best to follow the same pattern as `get_RBM_FLT_CALLEE_TRASH` -- it has the same flavor of "sometimes needs to be determined based on compilation information", and `RBM_FLT_CALLEE_TRASH` is defined to call that function. It will ensure that everyone gets the consistent view, but will of course result in a much less localized change. If it ends up requiring too many changes or being costly then I would be fine with what you suggest.

Reply:
Thanks for the suggestion! I gave that a try, and I think we use `RBM_INT_CALLEE_SAVED` in too many places to make this approach manageable; in particular, we use `RBM_INT_CALLEE_SAVED` in some static functions where we don't have access to the current `Compiler`'s state, so we can't easily figure out whether to include `RBM_SWIFT_ERROR` in the mask. Also, I think the additional check for `lvaSwiftErrorArg` would've been more expensive than necessary, since we check `RBM_INT_CALLEE_SAVED` (directly and indirectly, via other macros) in places irrelevant to method prolog/epilog generation.

I think `RegSet` provides a decent abstraction, though to make its usage consistent across the JIT, I had to make some trivial changes to architectures irrelevant to this PR.

Reply:
Ok, it seems fine to me for now. Some notes...

> and I think we use `RBM_INT_CALLEE_SAVED` in too many places to make this approach manageable

This doesn't exactly fill me with confidence. It means there are many places getting the wrong and inconsistent view after this PR. Are we sure that none of those need to be updated for correctness sake? If we hit a bug tail around this we should reconsider.

> Also, I think the additional check for `lvaSwiftErrorArg` would've been more expensive than necessary, since we check `RBM_INT_CALLEE_SAVED` (directly and indirectly, via other macros) in places irrelevant to method prolog/epilog generation.

I think we would design it similar to `get_RBM_FLT_CALLEE_TRASH` -- the mask is stored in `Compiler`, so it's just returning the field. There is no dynamic logic to compute it on every invocation.

Reply:
> This doesn't exactly fill me with confidence. It means there are many places getting the wrong and inconsistent view after this PR. Are we sure that none of those need to be updated for correctness sake? If we hit a bug tail around this we should reconsider.

This might be naive, but I think it's ok if we continue to think of `REG_SWIFT_ERROR` as callee-save elsewhere in the JIT, and limit the error register-specific quirks to prolog/epilog generation. We use `RBM_INT_CALLEE_SAVED` in the definition of `RBM_ALLINT` as well, so to ensure the latter includes `REG_SWIFT_ERROR`, we'd probably need to save the "normal" value of `RBM_INT_CALLEE_SAVED` to another macro -- the fact that this new macro would only be semantically different from `RBM_INT_CALLEE_SAVED` on Swift platforms seems messy. We'd also have to use this duplicated macro in the various static asserts that use `RBM_INT_CALLEE_SAVED` without access to compiler state.

The benefit of the current approach is its behavior is limited in scope. It might be too limited to the point where we are incorrectly treating `REG_SWIFT_ERROR` as callee-save (though I haven't run into that yet), but that seems easier to debug than the opposite scenario of making `RBM_INT_CALLEE_SAVED` stateful.